### PR TITLE
feat: added support for nested enum in request body in node

### PR DESCRIPTION
**Pull request by tiwarishubham635**
# Fixes #

Fixed the handling of nested enum objects in request body of node

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] Run `make test-docker`
- [x] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/resolver/node/NodeCodegenModelDataTypeResolver.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public void setCodegenModel(CodegenModelResolver codegenModelResolver) {
+        this.codegenModelResolver = codegenModelResolver;
+    }
+
+    @Override
+    public CodegenProperty resolve(CodegenProperty property, ApiResourceBuilder apiResourceBuilder) {
+        property =  super.resolve(property, apiResourceBuilder);
+        CodegenModel codegenModel = apiResourceBuilder.getModel(property.dataType);
+        if (codegenModel != null && !CodegenUtils.isPropertySchemaEnum(property) && !modelFormatMap.containsKey(property.dataType)) {
+            // this is recursion as codegenModelResolver will again call CsharpCodegenModelDataTypeResolver
```
Comment (kridai) [inline src/main/java/com/twilio/oai/resolver/node/NodeCodegenModelDataTypeResolver.java:37]: update this comment

File: src/main/java/com/twilio/oai/TwilioNodeGenerator.java [hunk 2]
```diff
@@ -102,16 +101,19 @@ public OperationsMap postProcessOperationsWithModels(final OperationsMap objs, L
...
+                                                                                    List.of(EnumConstants.NodeDataTypes.values()));
 
         NodeApiResourceBuilder nodeApiResourceBuilder = new NodeApiResourceBuilder(actionTemplate, opList, allModels, directoryStructureService
                 , twilioCodegen.getToggles(JSON_INGRESS));
 
         nodeApiResourceBuilder.updateApiPath()
             .updateTemplate()
-            .updateOperations(new NodeParameterResolver(conventionMapper, codegenModelResolver))
+            .updateOperations(new NodeParameterResolver(conventionMapper, nodeCodegenModelResolver))
             .updateResponseModel(new LanguagePropertyResolver(conventionMapper), codegenModelResolver);
```
Comment (kridai) [inline src/main/java/com/twilio/oai/TwilioNodeGenerator.java:117]: why are we not using nodeCodegenModelResolver here in updateResponseModel

File: src/main/java/com/twilio/oai/resolver/common/CodegenModelResolver.java [hunk 3]
```diff
@@ -45,26 +42,10 @@ public CodegenModel resolve(CodegenModel model, ApiResourceBuilder apiResourceBu
...
-                }
-                else {
-                    model.vendorExtensions.put(key, value);
-                }
-                return value;
-            });
         }
         return model;
     }
     public CodegenModel resolveNestedModel(CodegenProperty property, ApiResourceBuilder apiResourceBuilder) {
```
Comment (kridai) [inline src/main/java/com/twilio/oai/resolver/common/CodegenModelResolver.java:48]: remove this method if not being used

File: src/main/java/com/twilio/oai/resolver/node/NodeCodegenModelResolver.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                }
+                else {
+                    model.vendorExtensions.put(key, value);
+                }
+                return value;
+            });
+        }
+        return model;
+    }
+    @Override
```
Comment (kridai) [inline src/main/java/com/twilio/oai/resolver/node/NodeCodegenModelResolver.java:63]: no need to override, we can have this method private

File: src/main/java/com/twilio/oai/TwilioNodeGenerator.java (hunk 2 above)
Comment (tiwarishubham635) [reply src/main/java/com/twilio/oai/TwilioNodeGenerator.java:117]: So for request model, I have handled the resolving via Node specific resolvers, but for response model, it is handled in a different way, so it is getting overwritten wrongly. That's why I had to use common model resolver.

File: src/main/java/com/twilio/oai/resolver/common/CodegenModelResolver.java (hunk 3 above)
Comment (tiwarishubham635) [reply src/main/java/com/twilio/oai/resolver/common/CodegenModelResolver.java:48]: Are you referring the resolveNestedModel()?

**Review by kridai — APPROVED** [review]

**Pull request merged**